src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLease.cs

```diff
@@ -35,6 +35,23 @@ public DocumentServiceLease(DocumentServiceLease other)
         [JsonProperty("id")]
         public string Id { get; set; }
 
+        /// <summary>
+        /// Gets or sets property to be used as partition key path for lease collections created in GremlinAccounts
+        /// </summary>
+        [JsonProperty("gremlincompatid")]
+        public string GremlinCompatId
```

Comment:
What if the user has a property with this name? Wouldn't be better for the user to specify the pk they want to use?

Reply:
Note the problem you are specifying, exists even today before this PR. Default implementation assumes it can be only "id"

This is to address the default scenario where end user is using default lease management from this library. 
In this case, property names are controlled by this class and there would be no conflict.
In their custom lease management solution, it doesnt hit this path as I understand

Reply:
I understand, what I'm saying is that this fails with the same assumption that currently happens with `/id`. Giving the user the choice would be better suited.

Reply:
Let me give an example and you can tell me how it could work so that I can understand the proposal better
lets say I choose FooId as my partitionKey for the lease collection and created the collection. now when I pass it down to this library, what should the library do to the DocumentServiceLease class so that it can stamp the objects with FooId property.

I feel but will look at your reply for above example, What you are proposing would not possible as long lease implementation and its class definition is controlled by this library unless you are talking reflection and dynamically adding properties. I am not sure whether we want that kind of complexity, only reason customers complain about id as current partition property is because it doesnt work on gremlin. 

Ideally this library should create the lease collection since it controls definition of DocumentServiceLease and stores it in the lease collection. Right now there is an abstraction leak where customers discover that "id" should be the partitionkey because we look into the class definition/get runtime exception thrown saying what should be the partitionkey when DocumentServiceLease is internal.